File: LeetCode/206 Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param {ListNode} head
    # @return {ListNode}
    def reverseList(self, head):
        if head == None:
            return None
        previous = head
        current = head.next
        previous.next = None
        while current != None:
            next = current.next
            current.next = previous
            previous = current
            current = next
        return previous

File: LeetCode/206 Reverse Linked List.cpp
class Solution {
 public:
  ListNode *reverseList(ListNode *head) {
    ListNode *reversed_head = NULL;
    ListNode *current_node = head;
    while (current_node) {
      ListNode *temp = reversed_head;
      reversed_head = current_node;
      current_node = current_node->next;
      reversed_head->next = temp;
    }
    return reversed_head;
  }
};

int main() {
  return 0;
}
